fix: show bind preview on slash command hint, edit queries from there

AmanAgarwal041 opened the pull request:
## Description
This PR shows a preview of the slash command hints of suggested queries for binding. This is only limited to created actions for now. You can even edit the queries directly from here when the edit button appears on hover of the hint.
The binding appears in place of name of the query on hover

Notion link : https://www.notion.so/appsmith/Show-the-hovered-option-to-bind-the-preview-in-the-input-1307c5b2a6b048c9ae39ebb3dd6e2cae

<img width="384" alt="Screenshot 2024-03-25 at 3 10 52 PM" src="https://github.com/appsmithorg/appsmith/assets/7565635/68e2ce23-2cae-48b3-9d0e-a23f1693b33e">

Fixes #31112 

_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.JS"

### :mag: Cypress test results
<!-- This is an auto-generated comment: Cypress test results  -->
> [!CAUTION]  
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/8418035747>
> Commit: `356105f1dbe324f52505fd402be81ad84bfa6415`
> Cypress dashboard: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=8418035747&attempt=1&selectiontype=test&testsstatus=failed&specsstatus=fail" target="_blank"> Click here!</a>
> The following are new failures, please fix them before merging the PR: <ol>
> <li>cypress/e2e/Regression/ClientSide/Autocomplete/PropertyPaneSuggestion_spec.ts </ol>
> To know the list of identified flaky tests - <a href="https://internal.appsmith.com/app/cypress-dashboard/identified-flaky-tests-65890b3c81d7400d08fa9ee3?branch=master" target="_blank">Refer here</a>

<!-- end of auto-generated comment: Cypress test results  -->









<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced Quick Commands in the code editor with additional properties for suggestions and action switching functionality.
	- Added new CSS style for command suggestions in the code editor.
	- Updated the `BetaCard` component to accept a `className` prop for custom styling.
- **Refactor**
	- Renamed a function for retrieving plugin information to improve clarity and usage within the code editor component.
- **Chores**
	- Expanded imports in the code editor component to include various utilities and types for improved functionality.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

AmanAgarwal041 commented:
/build-deploy-preview skip-tests=true

AmanAgarwal041 commented:
/build-deploy-preview skip-tests=true recreate=true

AmanAgarwal041 commented on app/client/src/sagas/selectors.tsx:
```diff
@@ -233,3 +233,10 @@ export const getWidgetImmediateChildren = createSelector(
+    acc[p.id] = p;
+    return acc;
+  }, {}),
+);
```
Fixed

sneha122 commented:
@AmanAgarwal041 Can we please add unit test for this? Where we assert that on hover tooltip and the edit button is visible?

sneha122 commented:
/build-deploy-preview skip-tests=true

AmanAgarwal041 commented:
Add unit test case for the above solution : https://github.com/appsmithorg/appsmith/issues/32037

sneha122 commented:
@AmanAgarwal041 Tested the DP, looks good, just noticed one bug wrt to tooltips getting stuck, I know you had mentioned that it's a design system bug, but I think we should fix this because it reduces the quality of feature we are developing, whenever I click on any query, it gets stuck in the UI, added video in [slack thread](https://theappsmith.slack.com/archives/C04C1A774RJ/p1711101653639939)

AmanAgarwal041 commented:
/build-deploy-preview skip-tests=true recreate=true

AmanAgarwal041 commented:
> @AmanAgarwal041 Tested the DP, looks good, just noticed one bug wrt to tooltips getting stuck, I know you had mentioned that it's a design system bug, but I think we should fix this because it reduces the quality of feature we are developing, whenever I click on any query, it gets stuck in the UI, added video in [slack thread](https://theappsmith.slack.com/archives/C04C1A774RJ/p1711101653639939)

Fixed the above issue to replace tooltip solution to inplace binding text update

AmanAgarwal041 commented:
/build-deploy-preview skip-tests=true

AmanAgarwal041 commented:
Closing the pr due to : https://github.com/appsmithorg/appsmith/issues/31112#issuecomment-2018425604
